src/main/java/org/adoptopenjdk/maven/plugins/JsplitpkgscanMojo.java

```diff
@@ -1,70 +1,91 @@
 package org.adoptopenjdk.maven.plugins;
 
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.DefaultArtifact;
+import org.apache.maven.artifact.handler.DefaultArtifactHandler;
+import org.apache.maven.artifact.repository.ArtifactRepository;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.model.Dependency;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.artifact.ProjectArtifact;
 
+import javax.tools.Tool;
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.function.Consumer;
 
 /**
  * Goal which:
- *
+ * <p>
  * 1. touches a timestamp file. TODO remove this feature
  * 2. TODO Runs jsplitpkgscan
  */
 @Mojo(name = "jsplitpkgscan", defaultPhase = LifecyclePhase.PROCESS_SOURCES)
 public class JsplitpkgscanMojo extends AbstractMojo {
 
-    // Location of the file.
+    @Parameter(defaultValue = "${project}", readonly = true, required = true)
+    private MavenProject project;
+
+    @Parameter(defaultValue = "${session}", readonly = true, required = true)
+    private MavenSession session;
+
     @Parameter(defaultValue = "${project.build.directory}", property = "outputDir", required = true)
     private File outputDirectory;
 
     /**
      * Default execute method for Maven plugins.
+     *
      * @throws MojoExecutionException
      */
     @Override
     public void execute() throws MojoExecutionException {
-        // TODO We still create a text file until we can get jsplitpkgscan working correctly.
-        createTouchTxtFile();
-
         // Run jsplitpkgscan with the default arguments
-        // TODO figure out how to read in parameters set by the plugin configuraton
-        // runJsplitpkgscan()
-    }
+        for (Tool tool : ServiceLoader.load(Tool.class)) {
+            if ("jsplitpgkscan".equals(tool.toString())) {
+                List<String> artifactJars = new ArrayList<>();
+                //todo: add filter possibility to only include certain scopes
+                collectArtifacts(artifact -> artifactJars.add(artifact.getFile().getAbsolutePath()));
+
+                System.out.println(artifactJars);
```

Comment:
Let's add a TODO here to move this to a logger - that way folks could log to stdout or to somewhere else...